### PR TITLE
feat(events): add new hook && event `client.check_authn_complete`

### PR DESCRIPTION
**Pull request by lafirest**
Fixes EMQX-12135

Release version: v/e5.8.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_rule_engine/src/emqx_rule_events.erl [hunk 1]
```diff
@@ -1023,6 +1072,17 @@ columns_with_exam('client.check_authz_complete') ->
         {<<"timestamp">>, erlang:system_time(millisecond)},
         {<<"node">>, node()}
     ];
+columns_with_exam('client.check_authn_complete') ->
+    [
+        {<<"event">>, 'client.check_authz_complete'},
+        {<<"clientid">>, <<"c_emqx">>},
+        {<<"username">>, <<"u_emqx">>},
+        {<<"peerhost">>, <<"192.168.0.10">>},
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_events.erl:1080]: please add peername instead.
to unify, we should add peername to all events (which currently have peerhost), and consider peerhost deprecated.

**Comment by zmstone** [comment]
add a changelog.

File: apps/emqx_rule_engine/src/emqx_rule_api_schema.erl [hunk 2]
```diff
@@ -282,6 +282,17 @@ fields("ctx_check_authz_complete") ->
...
         {"result", sc(binary(), #{desc => ?DESC("event_result")})}
     ];
+fields("ctx_check_authn_complete") ->
+    Event = 'client.check_authn_complete',
+    [
+        {"event_type", event_type_sc(Event)},
+        {"event", event_sc(Event)},
+        {"clientid", sc(binary(), #{desc => ?DESC("event_clientid")})},
+        {"username", sc(binary(), #{desc => ?DESC("event_username")})},
+        {"peerhost", sc(binary(), #{desc => ?DESC("event_peerhost")})},
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_api_schema.erl:292]: peername ?

**Review by zmstone — APPROVED** [review]

File: apps/emqx_rule_engine/src/emqx_rule_events.erl [hunk 3]
```diff
@@ -854,6 +904,14 @@ test_columns('client.check_authz_complete') ->
         {<<"action">>, [<<"publish">>, <<"the action of publish or subscribe">>]},
         {<<"result">>, [<<"allow">>, <<"the authz check complete result">>]}
     ];
+test_columns('client.check_authn_complete') ->
+    [
+        {<<"clientid">>, [<<"c_emqx">>, <<"the clientid if the client">>]},
+        {<<"username">>, [<<"u_emqx">>, <<"the username if the client">>]},
+        {<<"reason_code">>, [<<"sucess">>, <<"the reason code">>]},
```
Comment (HJianBo) [inline apps/emqx_rule_engine/src/emqx_rule_events.erl:911]: typos

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2508)

**Pull request merged**